src/App.tsx

```diff
@@ -14,37 +14,85 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  None = '',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
 
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = [...goodsFromServer];
+
+  if (sortType === SortType.Alphabetical) {
+    visibleGoods.sort();
+  } else if (sortType === SortType.Length) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const reversedGoods = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const resetState = () => {
+    setSortType(SortType.None);
+    setIsReversed(false);
+  };
```

Comment:
So, if you decide to add explicit generic type, maybe you should add types here too? It's not a mistake and there is no need any changes, because TS can infer the correct types for you, especially in simple cases like this. Just note😄
```suggestion
  const reversedGoods: () => void = () => {
    setIsReversed(!isReversed);
  };

  const resetState: () => void = () => {
    setSortType(SortType.None);
    setIsReversed(false);
  };
```